Flaky test: OTel span tests fail when xdist co-locates them (reproducible on main with -n 3)

## Summary

Three OTel tests fail together when pytest-xdist happens to co-locate them on one worker:

```
FAILED tests/pytest/test_otel_reactive_execution.py::TestSpanHierarchy::test_calc_span_nested_under_reactive_update - assert 0 >= 2
FAILED tests/pytest/test_otel_reactive_flush.py::TestReactiveFlushInstrumentation::test_span_parent_child_relationship - assert 0 >= 2
FAILED tests/pytest/test_otel_reactive_flush.py::TestReactiveFlushInstrumentation::test_reactive_update_span_includes_session_id_from_context - AssertionError: reactive_update span should exist
```

**Job (attempt 1):** https://github.com/posit-dev/py-shiny/actions/runs/30473455629/job/90649155012 — `check (3.14, ubuntu-latest)`. **Passed on rerun.**

The exported spans list is empty (`0 = len([])`), i.e. no spans were recorded at all — consistent with another test in the same worker process having already installed a global tracer provider.

## This is reproducible on `main`, without any PR changes

The valuable part: this does **not** require a specific diff. Checking out `origin/main` clean and co-locating the OTel test files reproduces it by varying only the worker count:

```
$ uv run pytest tests/pytest/test_otel_logfire.py \
                tests/pytest/test_otel_reactive_flush.py \
                tests/pytest/test_otel_reactive_execution.py -q -n 2
70 passed, 2 skipped

$ ... -n 3
69 passed, 1 failed, 2 skipped

$ ... -n 4
69 passed, 1 failed, 2 skipped
```

So the trigger is which OTel tests land in the same worker process, not timing jitter.

## Why it surfaces on unrelated PRs

`pytest.ini` sets `addopts = ... --numprocesses auto`. With `--dist load`, the partition of tests across workers depends on the total collected test count — so **any PR that adds or removes a test can flip which OTel tests share a process**, with no causal connection to what the PR changed.

Observed exactly that twice: adding 3 unrelated unit tests to a PR turned the full local suite from 4/4 green into 2/4 runs failing on these same tests. Adding 3 *no-op* tests instead did not trigger it — the deciding factor is the resulting partition, not the content.

The suite already knows about this coupling. `test_otel_logfire.py` self-documents it by skipping:

```
SKIPPED [1] tests/pytest/test_otel_logfire.py:55: OpenTelemetry providers already configured (likely by another test in parallel execution)
```

That skip appears or not depending on worker count too (1 skipped at `-n auto`, 2 skipped at `-n 2/3/5/6` on clean `main`) — the same root cause, just landing on the more graceful branch. `test_otel_value_logging.py`'s session-scoped fixture docstring also calls out that OTel uses a global singleton provider and that `set_logger_provider()` cannot be called twice.

## Suggested direction

The tests that assert on span contents depend on owning the global tracer provider, but nothing enforces that ownership. Options, roughly in increasing robustness:

1. Give the span-asserting tests the same defensive skip `test_otel_logfire.py` uses — cheap, but converts failures into silently-skipped coverage.
2. Pin all OTel test files to one xdist worker (`--dist loadfile`/`loadgroup` via `@pytest.mark.xdist_group`), so co-location becomes deterministic rather than luck.
3. Stop relying on the process-global provider in these tests — use an isolated `TracerProvider` per test and read spans from its own exporter, instead of whatever the global happens to be.

(3) removes the coupling; (2) makes it deterministic and is a much smaller change.

<sub>AI-generated report. Failed on a PR whose diff was a `shiny/ui/_layout.py` refactor plus layout unit tests, unrelated to OTel.</sub>

--- a/tests/pytest/conftest.py
+++ b/tests/pytest/conftest.py
@@ -8,6 +8,9 @@ import pytest
 from opentelemetry.sdk.trace import TracerProvider
 from opentelemetry.sdk.trace.export.in_memory_span_exporter import InMemorySpanExporter
 
+from shiny.otel import _core
+from shiny.otel._constants import TRACER_NAME
+
 from .otel_helpers import otel_tracer_provider_impl
 
 
@@ -35,7 +38,7 @@ def _otel_tracer_provider_session() -> (
 @pytest.fixture
 def otel_tracer_provider(
     _otel_tracer_provider_session: Tuple[TracerProvider, InMemorySpanExporter],
-) -> Tuple[TracerProvider, InMemorySpanExporter]:
+) -> Iterator[Tuple[TracerProvider, InMemorySpanExporter]]:
     """
     Function-scoped pytest fixture for OpenTelemetry TracerProvider.
 
@@ -52,6 +55,16 @@ def otel_tracer_provider(
 
     This two-fixture pattern separates the one-time global setup (session)
     from the per-test cleanup (function), working correctly with pytest-xdist.
+
+    Why force `_core._tracer`?
+    --------------------------
+    `trace.set_tracer_provider()` is silently rejected when a provider was
+    already installed in this worker process (e.g. by `test_otel_logfire.py`
+    configuring logfire). In that case the session fixture's provider is not
+    the global one, spans go somewhere else, and the in-memory exporter stays
+    empty. Pointing Shiny's cached tracer at the test provider makes span
+    assertions independent of which tests happen to share the worker. This
+    mirrors what `test_otel_value_logging.py` does for `_core._logger`.
 
     Yields
     ------
@@ -80,4 +93,10 @@ def otel_tracer_provider(
     provider, exporter = _otel_tracer_provider_session
     # Clear spans from previous tests to ensure isolation
     exporter.clear()
-    return provider, exporter
+    # Force Shiny's cached tracer to come from the test provider, even if
+    # `trace.set_tracer_provider()` was rejected because another test already
+    # owned the global provider.
+    _core._tracer = provider.get_tracer(TRACER_NAME)
+    yield provider, exporter
+    # Reset so subsequent callers re-fetch from the global provider
+    _core._tracer = None

